### ISSUE TITLE
Inconsistent behavior when new stages are created in `Empirica.onStageEnded`

### ISSUE THREAD

**Issue by edwardneo** [empiricaly/empirica#342]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I'm trying to create a round with a flexible number of stages by creating stages when `Empirica.onStageEnded` is called.
```js
Empirica.onStageEnded(({ stage }) => {
  stage.round.addStage({
    name: "dummy",
    duration: 10000,
  });
});
```
Rather than generating me a round with infinite stages, it does one of two things:
1. It generates only one extra stage (two stages total) and ends the round without error.
2. After the initial stage, it gives a vague error on the server side and the experiment page is stuck on loading. The vague error is in the log output below.

### Steps To Reproduce

This repo has the bare bones setup to recreate the issue: [https://github.com/edwardneo/empirica-test](https://github.com/edwardneo/empirica-test)
It has a basic dummy stage with the `Empirica.onStageEnded` callback code provided above.

Create a basic batch with solo treatment and play through the game. It should give one of the errors mentioned above -- I'm not sure what causes one or the other to occur.

### Empirica Version

```text
Version: v1.4.0
SHA:     7865042
Branch:  main
Time:    2023-06-24T14:34:36Z

Client:  1.4.2
Server:  1.4.2
```
```


### What OS are you seeing the problem on?

macOS

### What browser are you seeing the problem on?

Chrome

### Relevant log output

```shell
[server] 15:02:56.091 ERR Error caught in "stage.start" callback:
[server] 15:02:56.162 ERR ZodError: [
[server] 15:02:56.162 ERR   {
[server] 15:02:56.162 ERR     "code": "invalid_type",
[server] 15:02:56.162 ERR     "expected": "string",
[server] 15:02:56.162 ERR     "received": "undefined",
[server] 15:02:56.162 ERR     "path": [],
[server] 15:02:56.162 ERR     "message": "Required"
[server] 15:02:56.162 ERR   }
[server] 15:02:56.162 ERR ]
[server] 15:02:56.162 ERR /Users/edwardneo/Documents/Coding/empirica/empirica-test/server/node_modules/zod/lib/index.mjs:449
[server] 15:02:56.162 ERR const error = new ZodError(ctx.common.issues);
[server] 15:02:56.162 ERR ^
[server] 15:02:56.162 ERR ZodError: [
[server] 15:02:56.162 ERR {
[server] 15:02:56.162 ERR "code": "invalid_type",
[server] 15:02:56.162 ERR "expected": "string",
[server] 15:02:56.162 ERR "received": "undefined",
[server] 15:02:56.162 ERR "path": [],
[server] 15:02:56.162 ERR "message": "Required"
[server] 15:02:56.162 ERR }
[server] 15:02:56.162 ERR ]
[server] 15:02:56.162 ERR   at m0 (/Users/edwardneo/Documents/Coding/empirica/empirica-test/server/node_modules/zod/lib/index.mjs:449:23)
[server] 15:02:56.162 ERR   at Ir.safeParse (/Users/edwardneo/Documents/Coding/empirica/empirica-test/server/node_modules/zod/lib/index.mjs:561:16)
[server] 15:02:56.162 ERR   at Ir.parse (/Users/edwardneo/Documents/Coding/empirica/empirica-test/server/node_modules/zod/lib/index.mjs:541:29)
[server] 15:02:56.162 ERR   at <anonymous> (/Users/edwardneo/Documents/Coding/empirica/empirica-test/server/node_modules/@empirica/core/src/admin/classic/classic.ts:598:24)
[server] 15:02:56.162 ERR   at Object.callback (/Users/edwardneo/Documents/Coding/empirica/empirica-test/server/node_modules/@empirica/core/src/admin/events.ts:88:11)
[server] 15:02:56.162 ERR   at Object.next (/Users/edwardneo/Documents/Coding/empirica/empirica-test/server/node_modules/@empirica/core/src/admin/cake.ts:307:30)
[server] 15:02:56.162 ERR   at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by npaton** [comment]
I have published a **prerelease** with a fix that might address your issue. I have not had the time to test on your specific problem, but it address a similar issue, so I figured I'd let you know if you wanted to try it out.

Since it's a **prerelease**, updating is a little different:

1. replace the contents of .empirica/release with version: v1.4.6
1. go into /client and run empirica npm install @empirica/core@next
1. go into /server and run the same npm command.

Let me know how it goes.

**Comment by npaton** [comment]
I pushed another fix testing this scenario. I think we should be good now. Please let me know.

**Comment by edwardneo** [comment]
Thanks! I tried running it again using version `v1.4.6` and the infinite loading doesn't occur anymore, but I'm still running into the first outcome where only one more stage is generated instead of infinitely generating. I pushed my version update to the Github repo linked above.

**Comment by npaton** [comment]
Ok, thanks! I will take a look at it as soon as possible.

**Comment by npaton** [comment]
My apologies for the delay. This should be fixed in the latest release (v1.6.1). Since we had an updater bug recently, you'll want to reinstall `empirica` with:

```
curl https://install.empirica.dev | sh
```

Then inside your project run:

```
empirica upgrade
```

Let me know if you are still seeing issues.
